src/cluster.ts

```diff
@@ -207,17 +282,26 @@ Please use the format 'my-cluster' or '${instance.name}/clusters/my-cluster'.`);
    * @example <caption>include:samples/document-snippets/cluster.js</caption>
    * region_tag:bigtable_get_cluster
    */
-  get(gaxOptions, callback) {
-    if (is.fn(gaxOptions)) {
-      callback = gaxOptions;
-      gaxOptions = {};
-    }
+  get(gaxOptionsOrCallback?: CallOptions|GetClusterCallback,
+      cb?: GetClusterCallback): void|Promise<GetClusterResponse> {
+    const callback =
+        typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
+    const gaxOptions =
+        typeof gaxOptionsOrCallback === 'object' && gaxOptionsOrCallback ?
+        gaxOptionsOrCallback :
+        {} as CallOptions;
 
-    this.getMetadata(gaxOptions, (err, metadata) => {
-      callback(err, err ? null : this, metadata);
-    });
+    this.getMetadata(
+        gaxOptions, (err?: ServiceError|null, metadata?: ICluster|null) => {
```

Comment:
I'm thinking of `ICluster` and other respective `I<class name>` interfaces as metadata - is this the correct way to think about it?